.github/workflows/ci.yml

```
@@ -261,6 +281,170 @@ jobs:
+  docker-e2e:
+    # Builds the runtime-local Dockerfile target from the current source
+    # (npm pack + COPY into image) and runs `gsd --version` inside the
+    # container. Catches missing system deps, broken postinstall, missing
+    # tarball files. Gated by detect-changes so doc-only PRs skip it.
+    # No untrusted GitHub event input is consumed in run: blocks.
+    timeout-minutes: 20
+    needs: detect-changes
+    if: >-
+      needs.detect-changes.outputs.docs-only != 'true' &&
+      needs.detect-changes.outputs.docker-changed == 'true'
+    runs-on: blacksmith-4vcpu-ubuntu-2404
```

actionlint (1.7.12)

[error] label "blacksmith-4vcpu-ubuntu-2404" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2025-vs2026", "windows-2022", "windows-11-arm", "ubuntu-slim", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "macos-latest", "macos-latest-xlarge", "macos-latest-large", "macos-26-intel", "macos-26-xlarge", "macos-26-large", "macos-26", "macos-15-intel", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xlarge", "macos-14-large", "macos-14", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file

(runner-label)